httpapi_plugins/ftd.py

```diff
@@ -113,13 +116,10 @@ def refresh_token_payload(refresh_token):
             raise AnsibleConnectionFailure('Username and password are required for login in absence of refresh token')
 
         url = self._get_api_token_path()
-        self._display(HTTPMethod.POST, 'login', url)
-
-        dummy, response_data = self._send_auth_request(
-            url, json.dumps(payload), method=HTTPMethod.POST, headers=BASE_HEADERS
-        )
-
-        response = self._response_to_json(self._get_response_value(response_data))
+        if url:
```

Comment:
It feels tangled that we call `_login_api_call` both from here and `_lookup_login_url`. How about encapsulating all these logic in one method iterating over token paths. E.g.,
```
def _send_login_request(self, payload):
    token_paths = [self._get_api_token_path()] if self._get_api_token_path() else self._get_known_token_paths()
    for token_path in token_paths:
            try:
                response = self._login_api_call(payload, url)
            except ConnectionError:
                display.vvvv('REST:request to {0} failed because of connection error'.format(url))
            else:
                self._set_api_token_path(url)
                return response

         raise ConnectionError("Token generation API was not found.")
```

Reply:
fixed